ui/apps/platform/src/Containers/Policies/Wizard/Step1/MitreTechniqueSelect.tsx

```diff
@@ -24,52 +34,128 @@ function MitreTechniqueSelect({
     mitreTechniques, // relevant techniques for tactic
     techniqueId,
 }: MitreTechniqueSelectProps): ReactElement {
-    const [isOpen, setIsOpen] = useState(false);
+    const { isOpen, setIsOpen, onSelect, onToggle } = useSelectState(handleSelectOption);
 
-    function onSelect(_event, optionId) {
-        setIsOpen(false);
-        handleSelectOption(optionId);
-    }
+    // Group techniques by base technique ID
+    const groupedTechniques = mitreTechniques.reduce(
+        (groups, technique) => {
+            const baseId = technique.id.includes('.') ? technique.id.split('.')[0] : technique.id;
+            const updatedGroups = { ...groups };
+            if (!updatedGroups[baseId]) {
+                updatedGroups[baseId] = [];
+            }
+            updatedGroups[baseId] = [...updatedGroups[baseId], technique];
+            return updatedGroups;
+        },
+        {} as Record<string, MitreTechnique[]>
+    );
+
+    // Find the display content for the selected value
+    const getDisplayContent = (): React.ReactNode => {
```

Comment:
Nit: Probably low impact here, but in general if we have a function that returns JSX its is more clear to make it a true React component. In this case it would require moving the definition out of render though.

Reply:
Yeah, that's reasonable. Since the Mitre-related components have their unique way of rendering things, maybe this is an opportunity to see if we can make it so the rendering can even be passed as a prop for more advanced cases 🤔 I will think about